org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaClientConnection.java

```diff
@@ -103,6 +125,21 @@ public void publishDiagnostics(PublishDiagnosticsParams diagnostics){
 		client.publishDiagnostics(diagnostics);
 	}
 
+
+	/**
+	 * Sends a message to client to apply the given workspace edit.
+	 * This is available since LSP v3.0 should be used
+	 * only by checking the ClientCapabilities.

```

Comment:
I have thought about it but we do not have a %100 compatible alternate functionality so I decided to let the caller decide on the alternate action.